// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.gate";
option java_outer_classname = "GateProto";
option objc_class_prefix = "RTG";

package CVImageService;

// Interface exported by the server.
service CVServer {
  //测试图像流传输，opencv实现编解码网络传输图像
  rpc CVMatImageStream(stream ImageMessage) returns (stream ImageMessage);

  //图像处理，客户端发送图像，服务器处理完成后返回
  rpc CVImageProcessFunction(stream ImageMessage) returns (stream ImageMessage);
}


//图像传信息，包含命令和图像信息
message ImageMessage {
  //命令
  string Command = 1;
 
  //行数
  int32 row = 2;
  //列数
  int32 colum = 3;  
  //类型
  int32 type = 4;

  //状态码
  int32 channel = 5;

  //图像buffer
  bytes EnCodeBuff = 6;
}

